### ISSUE TITLE
Enhancement: --version and --bugreport should report tag or commit hash for compiled Act instances

### ISSUE THREAD

**Issue by maxsu** [nektos/act#1069]
## Enhancement

When act is built from scratch or installed with `go install`, it would be helpful for the `--version` and `--bug-report` flags to report the tag and/or commit hash (and indicate act was compiled).

## Existing Behavior

 `--bug-report` leaves the version field empty, and the `--version` flag is not compiled:

```
$ act --version
Error: unknown flag: --version
```

```
$ act --bug-report
act version:
GOOS:                   linux
GOARCH:                 amd64
NumCPU:                 8
.
.
.
```

This leaves some room for surprise, and requires an workaround (see below).

## System information

Act version: 0.2.26
Install route: `go install github.com/nektos/act@latest`
OS: Pop_OS 21.10
Arch: x86-64

## Workaround
I can find the tag like so:

```bash
$ ls $GO_PATH/pkg/mod/github.com/nektos
act@v0.2.26
```

**Comment by maxsu** [comment]
Is the existing behavior designed to encourage people to file bug reports against official versions?

In that case, consider the requested feature makes it easier to comply with the issue template requirement:

    if you've built `act` yourself, please provide commit hash
    
There are good and bad sides to this:

```diff

Pros:
+ The feature makes it easier to report bugs
+ This may increase the rate of bug reports
+ Of those reports, more will be in compliance /w the issue template

Cons: 
- The feature lets inexperienced users comply /w the issue template
- ... So the additional issues may otherwise be of lower quality
```

**Comment by catthehacker** [comment]
It already provides version, but you have to supply it via ldflags

**Comment by maxsu** [comment]
@catthehacker thanks for the lead!

 I can now set Act to a very special version:
 
```bash
$ go install --ldflags "-X main.version=6.6.6.😈" github.com/nektos/act@latest
$ act --version
act version 6.6.6.😈
```

The project has two ways of auto-setting the version:

1. Manual builds via Makefile: `Makefile` lines [2](https://github.com/nektos/act/blob/master/Makefile#L2) and [26](https://github.com/nektos/act/blob/2bb3e74616f86ece6a11cd854ad54701542410fd/Makefile#L26) set the version as described in [this article](https://vincent.bernat.ch/en/blog/2019-makefile-build-golang). If the user is not at a tag, the logic in line 2 gives us a hash.

2. Github action: `.github/actions/release.yml` runs `goreleaser/goreleaser-action@v1`. That takes the argument `version: latest` and just does the right thing ([goreleaser/github.ts:L25-L27](https://github.com/goreleaser/goreleaser-action/blob/master/src/github.ts#L25-L27))

**Comment by maxsu** [comment]
@catthehacker I propose that `go install` should also set the correct version. `go install` is one of the documented install methods, so without this, less experienced users (like yours truly) end up with non-versioned installs.

**Comment by maxsu** [comment]
I thought about this, and realize I was projecting expectations set by boyter's [scc project](https://github.com/boyter/scc/):

1. `scc` hardcodes the version in their [CLI entrypoint](https://github.com/boyter/scc/blob/d46832f83af4b22466ea57261a13cc5b91363268/processor/processor.go#L22)
2. They increment the version through manual [version bump commits](https://github.com/boyter/scc/commit/d46832f83af4b22466ea57261a13cc5b91363268)
3. `scc` promotes `go install` as its first deployment option. 

You have implemented points 1 and 2 in PR #1070. Thank you!

You have also deprecated go install, in [wiki ad675d](https://github.com/nektos/act/wiki/Installation/_compare/41ad44b343fb60e31bb4985cb48daaecf16112c4...ad675d579a321ca59352bfb803678da62f4f977e) and #1123.

I will admit first class `go install` has a  special place in my heart as a promoter of go - Boyter's project is the reason I learned how to install go, and so I consider it an evangelist for go. But that alone does not mean my expectation is right. As you point out:
 
- supporting `go install` can introduce support overhead 
- `act` meets a much more complex need than `scc`, so its users are more likely to know `go install`. 
  - Moreover, PR 1070 solves the potential issue in the latter case

Therefore I am satisfied with your resolution. Thank you!
